bugfix in edifact.ts and unceStructureParser

MyOrk64 opened the pull request:
changed DTM Segment class for correct implementation of C509 DTMDateTimePeriod

MyOrk64 committed:
Update uneceStructurePageParser.ts

fixed missspelled 'detial' to 'detail'

MyOrk64 committed:
Update edifact.ts

bugfix streetAndNumberOrPostOfficeBoxIdentifier[n] is not data[0], but data[n]

MyOrk64 committed:
Update edifact.ts

changed DTM-Segment according to EDIFACT spec with subclass for dateTimePeriod as component C507.
(It is needed for my upcomming implemantation of writer class based on intechageBuilder. 
It popped up by checking the writer using edifact-data -in-> reader -> intercheangeBuilder -> validator -> writer -> edifact-data-out.
DTM -segment data was writeen as simple elements und not as elements of C507)

RovoMe commented on src/edifact.ts:
```diff
@@ -1067,25 +1067,34 @@ export class MessageDetails implements Segment {
-export class DateTimePeriod implements Segment {
-
-    tag = "DTM";
+class DateTimePeriod {
```
This name clashes with the one of the `DateTimePeriod` segment. 

```suggestion
class DateTimePeriodData {
```

to be in inline with the scheme used in the `edifact.ts` file

RovoMe commented on src/edifact.ts:
```diff
@@ -1067,25 +1067,34 @@ export class MessageDetails implements Segment {
+
+    tag = "DTM";
+
+    dateTimePeriod: DateTimePeriod;
```
This here needs to change then as well

```suggestion
    dateTimePeriod: DateTimePeriodData;
```

RovoMe commented on src/edifact.ts:
```diff
@@ -1067,25 +1067,34 @@ export class MessageDetails implements Segment {
+    dateTimePeriod: DateTimePeriod;
+
+    constructor(data: ResultType) {
+        this.dateTimePeriod = new DateTimePeriod(data.elements[0]);
```
... and here as well

```suggestion
        this.dateTimePeriod = new DateTimePeriodData(data.elements[0]);
```

MyOrk64 committed:
Update src/edifact.ts

Co-authored-by: Roman Vottner <rovo@gmx.at>

MyOrk64 committed:
Update src/edifact.ts

Co-authored-by: Roman Vottner <rovo@gmx.at>

MyOrk64 committed:
Update src/edifact.ts

Co-authored-by: Roman Vottner <rovo@gmx.at>

MyOrk64 committed:
bug in regex, elements names with (, )

added \\() to regex  to match 7124 "United Nations Dangerous Goods (UNDG) identifier"
i.e. DESADV_D01B
